sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/MesosResourcePool.java

```diff
@@ -18,15 +18,19 @@
 public class MesosResourcePool {
     private static final Logger logger = LoggerFactory.getLogger(MesosResourcePool.class);
 
-    private final Offer offer;
-    private final Map<String, List<MesosResource>> unreservedAtomicPool;
-    private final Map<String, Value> unreservedMergedPool;
-    private final Map<String, MesosResource> reservedPool;
+    private Offer offer;
+    private Map<String, List<MesosResource>> unreservedAtomicPool;
+    private Map<String, Value> unreservedMergedPool;
+    private Map<String, MesosResource> reservedPool;
 
     /**
      * Creates a new pool of resources based on what's available in the provided {@link Offer}.
      */
     public MesosResourcePool(Offer offer) {
+        init(offer);
+    }
+
+    private void init(Offer offer) {
```

Comment:
@nickbp:  I generally think the approach you outline is a good pattern, but I think mutating the ResourcePool is fine as it simplifies the pipeline and that is the intent of this object (that it be mutated).